### ISSUE TITLE
replace auth origin in runtime to enable usage with cloudlfare tunnels

### ISSUE THREAD

**Issue by Niki2k1** [sidebase/nuxt-auth#655]
### Describe the feature

Currently this lib does not work with the inbuilt `nuxi dev --tunnel` dev environment. Thats because the library needs to know the origin of the page.

I already tried getting the tunnel url with a hook like this:
```js
nuxt.hook('listen', async (_listenerServer, listener) => {
  const urls: ListenerURL[] = await listener.getURLs();
  const tunnelUrl = urls.find((url) => url.type === 'tunnel');

  if (tunnelUrl) {
    nuxt.options.runtimeConfig.baseURL = tunnelUrl.url; // <---- this does not seem to work
  }
});
```

But replacing the url at runtime does not seem to work.

Input would be appreciated, I could try to make an PR then.

**Comment by zoey-kaiser** [comment]
Hi @Niki2k1 👋 

As we are currently working on the rewrite to using authjs under the hood (which does work for cloudflare), I would therefore not prioritize this issue, as it may be resolved by https://github.com/sidebase/nuxt-auth/issues/673!

**Comment by Niki2k1** [comment]
@zoey-kaiser thats awesome, I'll check again when the rewrite to authjs is done.

**Comment by DavidDeSloovere** [comment]
@Niki2k1 You can patch the source code to make it work while we wait for a new version.
https://github.com/sidebase/nuxt-auth/issues/691#issuecomment-1971226750

**Comment by zoey-kaiser** [comment]
Closed in `0.10.0`: https://auth.sidebase.io/guide/advanced/url-resolutions
